spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java

```diff
@@ -331,6 +332,30 @@ public void unregister(@PathVariable("type") ApplicationType type, @PathVariable
 		this.unregister(type, name, appRegistration.getVersion());
 	}
 
+	@RequestMapping(method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void unregisterAll() {
+		List<AppRegistration> appRegistrations = appRegistryService.findAll();
+
+		for (AppRegistration appRegistration : appRegistrations) {
+			String applicationName = appRegistration.getName();
+			String applicationVersion = appRegistration.getVersion();
+			ApplicationType applicationType = appRegistration.getType();
+
+			if (applicationType != ApplicationType.task) {
+				String streamWithApp = findStreamContainingAppOf(applicationType, applicationName, applicationVersion);
+
+				if (streamWithApp != null) {
```

Comment:
Are we OK exiting with the exception without unregistering other apps that are not being used? If so, we could change the exception message to say something like unregister all cannot be done. 

or, We could possibly use `deleteAll(Iterable)` method from the `AppRegistrationRepository` to delete only the apps that are not being used.

Reply:
i copied the logic from `unregister` unless ive overlooked something here

Reply:
nm i get what your saying - dont stop unregistration if there are streams using apps; just dont unregister ones in use. i guess that may be a question for @sabbyanandan

Reply:
+1 to not removing the apps from the registry, if they are already in use. Maybe also hint that in the return message. Something like:

> The unregister operation is complete with an exception of a few apps that are associated with the stream definition.

Reply:
@sabbyanandan they don't get removed if they are in use. so seems like the options are:

a) do a blanket delete of apps **only** if none are assigned to streams, which is the current behavior - all or nothing.

b) delete apps that are not bound to streams, report back saying as you mentioned above something like "The unregister operation is complete with an exception of a few apps that are associated with the stream definition."

sounds like you are in favor of option, "b", right?

Reply:
Yeah, #b seems to be better from UX perspective.

Reply:
ok thx